### PR TITLE
Add SetInterpolators mutator, some BNS input file improvements

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

These are the last 3/4 commits needed for curved mesh FD.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/SetInterpolators.hpp [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+ */
+template <size_t Dim>
+struct SetInterpolators {
+  using return_tags = tmpl::list<
+      evolution::dg::subcell::Tags::InterpolatorsFromFdToNeighborFd<Dim>,
+      evolution::dg::subcell::Tags::InterpolatorsFromDgToNeighborFd<Dim>,
+      evolution::dg::subcell::Tags::InterpolatorsFromNeighborDgToFd<Dim>>;
+  using argument_tags =
+      tmpl::list<::domain::Tags::Element<Dim>, ::domain::Tags::Domain<Dim>,
+                 domain::Tags::Mesh<Dim>, domain::Tags::Mesh<Dim>,
```
Comment (kidder) [inline src/Evolution/DgSubcell/SetInterpolators.hpp:60]: why are you passing in the same tag twice?

Comment (nilsdeppe) [reply src/Evolution/DgSubcell/SetInterpolators.hpp:60]: In the future, when we have non-uniform h/p-refinement this needs to take neighbor meshes. That should be documented. I'll add a `warning` in the Dox and also a C++ comment in front of the arguments

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the review!

File: src/Evolution/DgSubcell/SetInterpolators.hpp [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            try {
+              tnsr::I<double, Dim, Frame::ElementLogical> logical_coord =
+                  map.inverse(extract_point(grid_coords, i));
+              for (size_t d = 0; d < Dim; ++d) {
+                logical_coords.get(d)[i] = logical_coord.get(d);
+              }
+            } catch (const std::bad_optional_access& e) {
+              ERROR(
+                  "Failed to get logical coordinates for neighbor's "
+                  "ghost "
```
Comment (knelli2) [inline src/Evolution/DgSubcell/SetInterpolators.hpp]: [optional] Strange string formatting

**Review by knelli2 — DISMISSED** [review]
Small thing. Squash if you choose to do it

File: src/Evolution/DgSubcell/SetInterpolators.hpp [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+ * elements. We will want to use high-order slope-limited FD interpolation in
+ * the future, but that requires neighbor communication. A slightly simpler
+ * approach would be to use high-order Lagrange interpolation, which still
+ * requires neighbor communication but does not require any additional changes
+ * to the reconstruction routines to work on non-uniform grids. This is what
+ * the Multipatch-MHD code does, relying on the slope limiting from the ghost
+ * zones to remove oscillations. I (Nils Deppe) am not sure I love that, but
+ * it's worth a try since it should be pretty easy to do.
+ *
+ * \warning Assumes all elements that do h/p-refinement won't go to FD. That
```
Comment (kidder) [inline src/Evolution/DgSubcell/SetInterpolators.hpp]: I would drop the first sentence of the warning as I found it confusing.  And then in the second sentence:
- "That is," -> "Currently assumes that"
- "refinement and level" -> "refinement level"
- "number of grid points" -> "dg_mesh and subcell_mesh"

Then if you want, add a sentence like "Thus it cannot be used with generic h/p-refinement"

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
done, rebased, squashed. Thanks for the reviews!

**Review by kidder — APPROVED** [review]

**Review by knelli2 — APPROVED** [review]

**Pull request merged**